pkg/tsdb/cloudwatch/response_parser.go

```diff
@@ -66,13 +67,42 @@ func parseGetMetricDataTimeSeries(metricDataResults map[string]*cloudwatch.Metri
 	result := tsdb.TimeSeriesSlice{}
 	partialData := false
 	metricDataResultLabels := make([]string, 0)
-	for k := range metricDataResults {
-		metricDataResultLabels = append(metricDataResultLabels, k)
+
+	if requestAzs, ok := query.Dimensions["AvailabilityZone"]; ok {
+
+		if len(requestAzs) > len(metricDataResults) {
+			// Data is missing, add empty metrics...
```

Comment:
This will not work in case the dimension in question has one value only.

Reply:
Can take a look at that!